packages/ibm-products/src/components/TagOverflow/TagOverflow.js

```diff
@@ -0,0 +1,266 @@
+import cx from 'classnames';
+
+import { getDevtoolsProps } from '../../global/js/utils/devtools';
+import { pkg } from '../../settings';
+
+// Carbon and package components we use.
+import { Tag, Tooltip } from '@carbon/react';
+import { TagSet } from '../TagSet';
+import { useResizeObserver } from '../../global/js/hooks/useResizeObserver';
+
+// The block part of our conventional BEM class names (blockClass__E--M).
+const blockClass = `${pkg.prefix}--tag-overflow`;
+const componentName = 'TagOverflow';
+
+// NOTE: the component SCSS is not imported here: it is rolled up separately.
+
+// Default values can be included here and then assigned to the prop params,
+// e.g. prop = defaults.prop,
+// This gathers default values together neatly and ensures non-primitive
+// values are initialized early to avoid react making unnecessary re-renders.
+// Note that default values are not required for props that are 'required',
+// nor for props where the component can apply undefined values reasonably.
+// Default values should be provided when the component needs to make a choice
+// or assumption when a prop is not supplied.
+
+// Default values for props
+// const defaults = {
+//   /* TODO: add defaults for relevant props if needed */
+// };
+
+/**
+ * TODO: A description of the component.
+ */
+export let TagOverflow = React.forwardRef(
+  (
+    {
+      // The component props, in alphabetical order (for consistency).
+      className,
+      items,
+      itemTemplate,
+      maxVisible,
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
+  ) => {
+    // type of Tags elements within the component
+    const tagType = 'blue';
+
+    const localContainerRef = useRef(null);
+    const containerRef = ref || localContainerRef;
+    const itemRefs = useRef(null);
+    const overflowRef = useRef(null);
+    // measurementOffset is the value of margin applied on each items,
+    // this value is required for calculating how many items will fit within the container
+    const measurementOffset = 4;
+    const overflowIndicatorWidth = 40;
+
+    const [containerWidth, setContainerWidth] = useState(0);
+    const [visibleItems, setVisibleItems] = useState([]);
+    const [overflowItems, setOverflowItems] = useState([]);
+
+    const handleResize = () => {
+      setContainerWidth(containerRef.current.offsetWidth);
+    };
+
+    useResizeObserver(containerRef, handleResize);
+
+    const getMap = () => {
+      if (!itemRefs.current) {
+        itemRefs.current = new Map();
+      }
+      return itemRefs.current;
+    };
+
+    const itemRefHandler = (id, node) => {
+      const map = getMap();
+      if (id && node && map.get(id) !== node.offsetWidth) {
+        map.set(id, node.offsetWidth);
+      }
+    };
+
+    const getVisibleItems = useCallback(() => {
+      if (!itemRefs.current) {
+        return items;
+      }
+
+      const map = getMap();
+      const overflowContanerWidth =
```

Comment:
small typo. should be `overflowContainerWidth`. that should probably have been caught by the spellchecker!